### ISSUE TITLE
camerapreview plugin fix

### ISSUE THREAD

**Issue by maxborgatti** [mapsplugin/cordova-plugin-googlemaps#1784]
editing this plugin's code will fix an issue on camerapreview's plugin
https://github.com/cordova-plugin-camera-preview/cordova-plugin-camera-preview/issues/369

can this be confirmed by someone else?

**Comment by wf9a5m75** [comment]
Please share your project files on github.
Since this plugin handles the touch events, it seems this modification breaks this plugin behavior.

If you share your project files (or demo project files), I will check it. Otherwise, I don't support it. (Usually I don't support the compatibility with other plugins

**Comment by maxborgatti** [comment]
i think that already is a demo:
https://github.com/cordova-plugin-camera-preview/cordova-plugin-camera-preview-sample-app

i use ionic3/ang4 but i think it will be the same: adding cordova-plugin-googlemaps will prevent the html to stay above camera

![schermata 2017-09-29 alle 13 03 12](https://user-images.githubusercontent.com/8874135/31013505-affaf4a2-a516-11e7-8e37-e8d61adade1a.png)

**Comment by wf9a5m75** [comment]
No. I asked the demo app including the camera page and the map page both.

**Comment by wf9a5m75** [comment]
ping.

**Comment by maxborgatti** [comment]
https://github.com/maxborgatti/camerapreview-vs-maps

use your api keys in config:
`    <plugin name="cordova-plugin-googlemaps" spec="https://github.com/mapsplugin/cordova-plugin-googlemaps#multiple_maps">
        <variable name="API_KEY_FOR_ANDROID" value="YOUR_API_KEY" />
        <variable name="API_KEY_FOR_IOS" value="YOUR_API_KEY" />
    </plugin>`

comment the camerapreview code if you want to see the html below
`      this.cameraPreview.startCamera(this.cameraPreviewOpts).then(
        (res) => {
          console.log(res);
        },
        (err) => {
          console.log(err);
      });
      this.cameraPreview.show();`

the camera should go below html 
`    toBack: true,`

but it does not until you remove the map plugin

**Comment by wf9a5m75** [comment]
This is the screen record of just built your app.
![capture](https://user-images.githubusercontent.com/167831/31303442-836d1cf6-aac2-11e7-80e6-c01dc65a679d.gif)

**Comment by maxborgatti** [comment]
The demo is ok, and i hope the issue is clear also to you: camera stays on top, ignoring toBack value.

If you remove maps plugin and make bg transparent (i think i'd already set scss like that) you'll se the bug disappear and the camera going back html

**Comment by wf9a5m75** [comment]
The camera plugin and this plugin changes the view order(hierarchy). The both manipulate the click event with the same approach. That's why the both plugin are conflict.
This is not a bug.

You need to remove the map when you want to open the camera.

**Comment by maxborgatti** [comment]
i did update repo, commenting out map. you can pull and test. nothing changes.

removing plugin like this, fixes (the camera goes back html):

![image](https://user-images.githubusercontent.com/8874135/31311018-d79cf4a4-aba3-11e7-9e70-e4f604eaa121.png)

am i missing something?

**Comment by wf9a5m75** [comment]
No. What I mean is you need to execute `map.remove()` when you oepn the canera

**Comment by maxborgatti** [comment]
no effects

![image](https://user-images.githubusercontent.com/8874135/31311198-7a6e945a-aba7-11e7-892c-94d1e5ad8fd9.png)

please feel free to edit demo code your own. also, did you take a look to my suggestion? 
https://github.com/cordova-plugin-camera-preview/cordova-plugin-camera-preview/issues/369

**Comment by wf9a5m75** [comment]
Your suggested way breaks this plugin (did you read my commnet?).

**Comment by wf9a5m75** [comment]
Unfortunately this plugin and the camera plugin are conflict. You need to give up one of them.

**Comment by wf9a5m75** [comment]
Fortunately, I fixed this problem on another issue fixing.

https://github.com/wf9a5m75/camerapreview-vs-maps

![demo](https://user-images.githubusercontent.com/167831/34397638-8f61219c-eb2c-11e7-94fe-907251b80918.gif)
